quantum/via.c

```diff
@@ -495,3 +514,86 @@ void via_qmk_rgblight_set_value(uint8_t *data) {
 }
 
 #endif // #if defined(VIA_QMK_RGBLIGHT_ENABLE)
+
+#if defined(VIA_QMK_RGB_MATRIX_ENABLE)
+
+// VIA supports only 4 discrete values for effect speed; map these to some
+// useful speed values for RGB Matrix.
+enum speed_values {
+    RGBLIGHT_SPEED_0 = UINT8_MAX / 16, // not 0 to avoid really slow effects
+    RGBLIGHT_SPEED_1 = UINT8_MAX / 4,
+    RGBLIGHT_SPEED_2 = UINT8_MAX / 2,     // matches the default value
+    RGBLIGHT_SPEED_3 = UINT8_MAX / 4 * 3, // UINT8_MAX is really fast
+};
+
+static uint8_t speed_from_rgblight(uint8_t rgblight_speed) {
+    switch (rgblight_speed) {
+        case 0:
+            return RGBLIGHT_SPEED_0;
+        case 1:
+            return RGBLIGHT_SPEED_1;
+        case 2:
+        default:
+            return RGBLIGHT_SPEED_2;
+        case 3:
+            return RGBLIGHT_SPEED_3;
+    }
+}
+
+static uint8_t speed_to_rgblight(uint8_t rgb_matrix_speed) {
+    if (rgb_matrix_speed < ((RGBLIGHT_SPEED_0 + RGBLIGHT_SPEED_1) / 2)) {
+        return 0;
+    } else if (rgb_matrix_speed < ((RGBLIGHT_SPEED_1 + RGBLIGHT_SPEED_2) / 2)) {
+        return 1;
+    } else if (rgb_matrix_speed < ((RGBLIGHT_SPEED_2 + RGBLIGHT_SPEED_3) / 2)) {
+        return 2;
+    } else {
+        return 3;
+    }
+}
+
+void via_qmk_rgb_matrix_get_value(uint8_t *data) {
+    uint8_t *value_id   = &(data[0]);
+    uint8_t *value_data = &(data[1]);
+    switch (*value_id) {
+        case id_qmk_rgblight_brightness:
+            value_data[0] = rgb_matrix_get_val();
+            break;
+        case id_qmk_rgblight_effect:
+            value_data[0] = rgb_matrix_get_mode();
+            break;
+        case id_qmk_rgblight_effect_speed:
+            value_data[0] = speed_to_rgblight(rgb_matrix_get_speed());
+            break;
+        case id_qmk_rgblight_color:
+            value_data[0] = rgb_matrix_get_hue();
+            value_data[1] = rgb_matrix_get_sat();
+            break;
+    }
```

Comment:
no default?

Reply:
It's mostly copy-pasta, and those didn't include it.  Any other values aren't technically supported, either.

But I can add a default, if need be.

Reply:
Shouldn't matter -- the buffer is preallocated and shouldn't be messed with if unused.
All we'd end up with is `default: break;`.